CMakeLists.txt

```diff
@@ -182,9 +184,9 @@ add_subdirectory(src)
 # needed for openal.pc.in and openal-config.in
 set(prefix ${CMAKE_INSTALL_PREFIX})
 set(exec_prefix "\${prefix}")
-set(libdir "\${exec_prefix}/lib${LIB_SUFFIX}")
-set(bindir "\${exec_prefix}/bin")
-set(includedir "\${prefix}/include")
+set(libdir ${CMAKE_INSTALL_FULL_LIBDIR})
+set(bindir ${CMAKE_INSTALL_FULL_BINDIR})
+set(includedir ${CMAKE_INSTALL_FULL_INCLUDEDIR})
```

Comment:
I'm not sure that `CMAKE_INSTALL_PREFIX` cane be changed at install time, but that's because I always follow the following flow:
1. Make build dir (e.g. `_build`)
2. Run cmake including `-DCMAKE_INSTALL_PREFIX=/usr`
3. Run `make install DESTDIR=<path-to-build-jail>`

I'm not sure how you would change the system prefix during step 3.

Reply:
The reason I would use `${CMAKE_INSTALL_FULL_INCLUDEDIR}` instead of, say `${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}` is that GNUInstallDirs allows you to specify an absolute path for each of the options it supports.

For instance: `cmake -DCMAKE_INSTALL_INCLUDEDIR=/custom/path` results in:
- CMAKE_INSTALL_PREFIX=/usr
- CMAKE_INSTALL_INCLUDEDIR=/custom/path
- CMAKE_INSTALL_FULL_INCLUDEDIR=/custom/path

Whereas when it's a non-absolute path: `cmake -DCMAKE_INSTALL_INCLUDEDIR=libfoo` results in:
- CMAKE_INSTALL_PREFIX=/usr
- CMAKE_INSTALL_INCLUDEDIR=libfoo
- CMAKE_INSTALL_FULL_INCLUDEDIR=/usr/libfoo

Reply:
Yeah, I was under the impression that best practice was to use a relative path in those places. In any case, I'll merge this and then tweak it a bit myself. Thanks!